lib/dues_calculator.rb

```diff
@@ -9,16 +9,20 @@ def self.dues_per_competitor(country_iso2, base_entry_fee_lowest_denomination, c
   end
 
   def self.dues_per_competitor_in_usd(country_iso2, base_entry_fee_lowest_denomination, currency_code)
```

Comment:
You're doing an awful lot of `&.` and `|| 0` safe navigation throughout the whole method. Most of if seems to be related to the fact that `country_band_detail` can be null. When and under which circumstances can this happen? And if it happens, can you just `return` early and then after the return be sure that the CB details exist?

Reply:
> When and under which circumstances can this happen

I think it happens only if there is no active country_band_detail for a country.

I've returned early as you suggested.